src/file_filter.rs

```diff
@@ -58,12 +51,28 @@ pub fn is_valid_file(path: &PathBuf, allow_hidden_folders: bool, ignore_size_thr
     Ok(())
 }
 
-pub fn is_this_inside_blacklisted_dir(path: &PathBuf) -> bool {
+pub fn is_this_inside_blacklisted_dir(path: &PathBuf, privacy_settings: &FilePrivacySettings) -> bool {
     let mut path = path.clone();
     while path.parent().is_some() {
         path = path.parent().unwrap().to_path_buf();
-        if let Some(file_name) = path.file_name() {
-            if BLACKLISTED_DIRS.contains(&file_name.to_str().unwrap_or_default()) {
+        if is_this_blacklisted_file(&path, &privacy_settings) {
+            return true;
+        }
+    }
+    false
```

Comment:
the default must be true (access denied), if privacy_settings is default, as in load_privacy_rules_if_needed_weak() if it can't load the privacy rules